lib/Utils/PlatformWin.cpp

```
@@ -0,0 +1,687 @@
+           index++) {
+        const char *sp = keyName;
+        while (*sp && !isdigit(*sp))
+          sp++;
+        if (!*sp)
+          continue;
+        const char *ep = sp + 1;
+        while (*ep && (isdigit(*ep) || (*ep == '.')))
+          ep++;
+        char numBuf[32];
+        ::strncpy(numBuf, sp, sizeof(numBuf) - 1);
+        numBuf[sizeof(numBuf) - 1] = '\0';
+        double dvalue = ::strtod(numBuf, NULL);
+        if (dvalue > bestValue) {
+          // Test that InstallDir is indeed there before keeping this index.
+          // Open the chosen key path remainder.
+          ::strcpy(bestName, keyName);
+          // Append rest of key.
+          ::strncat(bestName, nextKey, sizeof(bestName) - 1);
+          bestName[sizeof(bestName) - 1] = '\0';
+          lResult = ::RegOpenKeyExA(hTopKey, bestName, 0,
+                                    KEY_READ | KEY_WOW64_32KEY, &hKey);
+          if (lResult == ERROR_SUCCESS) {
+            if (readFullStringValue(hKey, valueName, outValue)) {
+              // bestIndex = (int)index;
+              bestValue = dvalue;
+              returnValue = true;
+            }
+            ::RegCloseKey(hKey);
+          }
+        }
+        size = sizeof(keyName) - 1;
+      }
+      ::RegCloseKey(hTopKey);
+    } else
+      ReportError(lResult, "RegOpenKeyEx");
+  } else {
+    // If subKey is empty, then valueName is subkey, and we retreive that
+    if (subKey[0]==0) {
+      subKey = valueName;
+      valueName = nullptr;
+    }
+    lResult = ::RegOpenKeyExA(hRootKey, subKey, 0, KEY_READ | KEY_WOW64_32KEY,
+                              &hKey);
+    if (lResult == ERROR_SUCCESS) {
+      returnValue = readFullStringValue(hKey, valueName, outValue);
+      ::RegCloseKey(hKey);
+    } else
+      ReportError(lResult, "RegOpenKeyEx");
+  }
+  return returnValue;
+}
+
+int GetVisualStudioVersionCompiledWith() {
+#if (_MSC_VER >= 1900)
+  return 14;
```

Can we please just error out instead of guessing possibly the wrong install? Debugging a failure here (i.e. a case where we pick up the wrong reg keys of a side-by-side install of different VC versions) will be a nightmare. I'd prefer to error out for new versions and then add them when they arrive. New VC happens one a year, we can cope with that.